src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -1085,4 +1126,81 @@ mod tests {
             .tests(30)
             .quickcheck(static_header as fn(u64, u32) -> TestResult);
     }
+
+    #[test]
+    fn bda_test_recovery() {
+        fn corrupt_byte<F>(f: &mut F, position: u64, mask: u8) -> Result<()>
+        where
+            F: Read + Seek + SyncAll,
+        {
+            let mut byte_to_corrupt = [0; 1];
+            f.seek(SeekFrom::Start(position))?;
+            f.read(&mut byte_to_corrupt)?;
+            byte_to_corrupt[0] ^= mask;
```

Comment:
Might as well not take any ```mask``` parameter and instead use bitwise negation, ```!```, I would think.

Reply:
Done, thanks.  Originally I was planning on walking through different values of mask.